src/network/taraxa_capability.cpp

```diff
@@ -122,35 +122,53 @@ void TaraxaCapability::sealAndSend(NodeID const &nodeID, unsigned packet_type, R
   });
 }
 
-std::pair<bool, blk_hash_t> TaraxaCapability::checkDagBlockValidation(DagBlock const &block) {
+std::pair<bool, std::vector<blk_hash_t>> TaraxaCapability::checkDagBlockValidation(DagBlock const &block) {
+  std::vector<blk_hash_t> missing_blks;
+
   if (dag_blk_mgr_->getDagBlock(block.getHash())) {
     // The DAG block exist
-    return std::make_pair(true, blk_hash_t());
+    return std::make_pair(true, missing_blks);
   }
 
   level_t expected_level = 0;
   for (auto const &tip : block.getTips()) {
     auto tip_block = dag_blk_mgr_->getDagBlock(tip);
     if (!tip_block) {
       LOG(log_nf_dag_sync_) << "Block " << block.getHash().toString() << " has a missing tip " << tip.toString();
-      return std::make_pair(false, tip);
+      missing_blks.push_back(tip);
+    } else {
+      expected_level = std::max(tip_block->getLevel(), expected_level);
     }
-    expected_level = std::max(tip_block->getLevel(), expected_level);
   }
+
   auto pivot = block.getPivot();
   auto pivot_block = dag_blk_mgr_->getDagBlock(pivot);
   if (!pivot_block) {
-    LOG(log_nf_) << "Block " << block.getHash().toString() << " has a missing pivot " << pivot.toString();
-    return std::make_pair(false, pivot);
+    LOG(log_nf_dag_sync_) << "Block " << block.getHash().toString() << " has a missing pivot " << pivot.toString();
```

Comment:
Why info and not error or warning at least ?

Reply:
I think later is it logged as warning

Reply:
What does it matter ? If something is warning/error it should be logged with appropriate level everywhere